Class-19-(Authentication).md

```diff
@@ -0,0 +1,25 @@
+# Class 19: Authentication
+
+## In the prep work this week, the students learned
+
+1. What authentication is.
+1. What sessions and cookies are and how sessions and cookies are used to verify a user's identity.
+1. The importance of encryption.
+1. Why developers use one-way encryption to protect users' passwords.
+1. How to use Spring Security in an application to enable authentication.
+
+## Announcements
+
+1. Students are almost done with the class! One class left!
+1. Assignment #4 is due next class! 
+1. Check with your course manager for any additional announcements.
+
+## Large Group Time (Instructor)
+
+### Class 19 Topics That Require Careful Attention
+1. Review hashing and one-way encryption. Remind students that passwords are not stored, but the encrypted password is stored.
+1. 
```

Comment:
some ideas:
- discuss popular hashing algorithms and how to implement them
- cover salting as a practice - what is it and why it helps 
- a lot of this section of the book covers a tutorial walkthrough of setting up a basic user authentication work flow in coding-events. there are not videos of this process, nor do we present the finished product. Lots of information is delivered in the tutorial, including some new Spring packages and advanced topics that we only introduce briefly. It is important to emphasize for students that it's ok to treat this topic as an introduction. We are not asking them to be security experts at this point, but this is all valuable information for professional applications and getting exposed to it early on is very useful for their careers
- Emphasize the importance of the process, not necessarily knowing the exact syntax to get it done - syntax is easier to google or research!
- We ask the students to add the same auth flow to techjobs for studio work, so they get a chance to practice this again

Reply:
Thank you!